koopmans/workflows/_workflow.py

```diff
@@ -609,6 +612,29 @@ def load_old_calculator(self, qe_calc):
 
         return old_calc.is_complete()
 
+    def check_convergence(self, qe_calc):
+        # Check the convergence of a given calculation
+        if isinstance(qe_calc, (calculators.KoopmansScreenCalculator, calculators.KoopmansHamCalculator)):
+            # is_converged not implemented yet for KoopmansScreenCalculator and KoopmansHamCalculator
+            pass
+        elif qe_calc.is_converged():
+            if isinstance(qe_calc, calculators.KoopmansCPCalculator):
+                # Check spin-up and spin-down eigenvalues match
+                if 'eigenvalues' in qe_calc.results and qe_calc.parameters.do_outerloop \
+                        and qe_calc.parameters.nspin == 2 and qe_calc.parameters.tot_magnetization == 0 \
+                        and not qe_calc.parameters.fixed_state and len(qe_calc.results['eigenvalues']) > 0:
+                    rms_eigenval_difference = np.sqrt(np.mean(np.diff(qe_calc.results['eigenvalues'], axis=0)**2))
+                    if rms_eigenval_difference > 0.05:
+                        utils.warn('Spin-up and spin-down eigenvalues differ substantially')
+        else:
+            if isinstance(qe_calc, calculators.Wannier90Calculator):
+                # For projwfs and preproc calculations the convergence check cannot be applied;
+                # for mlwfs a warning is printed out in case the calculation is not converged
+                if self.parameters.init_orbitals != 'projwfs' and 'preproc' not in qe_calc.prefix:
+                    utils.warn(f'Be careful, the wannierization did not converge !')
+            else:
+                raise CalculationFailed(f'{qe_calc.prefix} is not converged')
+
```

Comment:
I know this is how I told you to do it, but I think we need to tweak this a bit. This code has a lot of overlap with `Calculator.is_converged()` and it's a bit confusing what this code is doing differently to `calc.is_converged()`. Also some of the above isn't really "checking convergence" it is "checking convergence as well as doing a few other sanity checks".

Whenever you end up writing code like this (where you are performing a lot of `isinstance` checks) it is nicer to try and give every calculator a method (e.g. `calc.postcalc_checks()`) and then simply we call `calc.postcalc_checks()` instead of doing something different for different calculator classes. That way, the logic associated with a calculator is contained within the calculator class and not within the workflow class.

What makes this tricky here is that we currently make reference to `self.parameters.init_orbitals`...

Maybe can we...
1) rename this function so it doesn't just appear to be about convergence e.g. "run_postcalc_checks"
2) move as much of the calculator-specific logic here into the associated calculator classes

I'm to discuss more in person about specifics.